### ISSUE TITLE
Detect duplicate types

### ISSUE THREAD

**Issue by jl-wynen** [scipp/sciline#138]
https://github.com/scipp/esssans/pull/102 had a duplicate type. This can break pipelines. Can we detect them automatically somehow? Maybe we can use a heuristic to search for them in source code or hook into a linter to list all type definitions. This could be hooked into pre-commit.

**Comment by SimonHeybrock** [comment]
Can you clarify what is duplicate and what broke? Having types with identical names in different modules sounds like a valid thing?

**Comment by jl-wynen** [comment]
See the linked PR. There were two identically named types and some providers used one, other used the other. This meant that those providers did not connect into a pipeline as expected.

**Comment by SimonHeybrock** [comment]
Yes, but can you clarify what Sciline could/should do about this? Isn't it a valid application to have types of the same naming in different packages or modules?

**Comment by jl-wynen** [comment]
Valid but confusing.
But note that this issue is about types within a single project. I suggested check using an external tool, not in Sciline. This would prevent using the same name in different modules in the same package which is almost definitely a mistake.

**Comment by SimonHeybrock** [comment]
Makes sense. Do you want to transfer the issue to another project then?
